src/types.ts

```diff
@@ -25,3 +25,25 @@ export enum SelectionType {
   Hide,
   Highlight,
 }
+
+export enum AgentLevel {
+  Agent,
+  ChildAgent,
+}
+
+interface BaseCustomCheckboxProps {
+  agentName: string;
+  agentLevel: AgentLevel;
+}
+
+export interface ParentAgentProps extends BaseCustomCheckboxProps {
+  agentLevel: AgentLevel.Agent;
+  childName?: never;
+}
+
+export interface ChildAgentProps extends BaseCustomCheckboxProps {
+  agentLevel: AgentLevel.ChildAgent;
+  childName: string;
+}
```

Comment:
```suggestion
export interface ChildCheckboxProps extends BaseCustomCheckboxProps {
  level: Level.Child;
  parentName: string;
}
```

then the main name would be the child name. From the point of view of this node, its name is just its name. It has a parent.